bundles/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/camera/CameraHandler.java

```diff
@@ -0,0 +1,144 @@
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.openhab.binding.netatmo.internal.handler.NetatmoModuleHandler;
+
+import static org.openhab.binding.netatmo.internal.ChannelTypeUtils.toOnOffType;
+import static org.openhab.binding.netatmo.internal.ChannelTypeUtils.toStringType;
+import static org.openhab.binding.netatmo.internal.NetatmoBindingConstants.*;
+
+/**
+ * {@link CameraHandler} is the class used to handle Camera Data
+ *
+ * @author Sven Strohschein (partly moved code from NAWelcomeCameraHandler to introduce inheritance, see NAWelcomeCameraHandler)
+ *
+ */
+public class CameraHandler extends NetatmoModuleHandler<NAWelcomeCamera> {
+
+    private static final String LIVE_PICTURE = "/live/snapshot_720.jpg";
+
+    public CameraHandler(@NonNull Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected void updateProperties(NAWelcomeCamera moduleData) {
+        updateProperties(null, moduleData.getType());
+    }
+
+    @SuppressWarnings("null")
+    @Override
+    protected State getNAThingProperty(String channelId) {
+        switch (channelId) {
+            case CHANNEL_CAMERA_STATUS:
+                return getStatusState();
+            case CHANNEL_CAMERA_SDSTATUS:
+                return getSdStatusState();
+            case CHANNEL_CAMERA_ALIMSTATUS:
+                return getAlimStatusState();
+            case CHANNEL_CAMERA_ISLOCAL:
+                return getIsLocalState();
+            case CHANNEL_CAMERA_LIVEPICTURE_URL:
+                return getLivePictureURLState();
+            case CHANNEL_CAMERA_LIVEPICTURE:
+                return getLivePictureState();
+            case CHANNEL_CAMERA_LIVESTREAM_URL:
+                return getLiveStreamState();
+        }
+        return super.getNAThingProperty(channelId);
+    }
+
+    protected State getStatusState() {
+        return module != null ? toOnOffType(module.getStatus()) : UnDefType.UNDEF;
+    }
+
+    protected State getSdStatusState() {
+        return module != null ? toOnOffType(module.getSdStatus()) : UnDefType.UNDEF;
+    }
+
+    protected State getAlimStatusState() {
+        return module != null ? toOnOffType(module.getAlimStatus()) : UnDefType.UNDEF;
+    }
+
+    protected State getIsLocalState() {
+        return module != null ? toOnOffType(module.getIsLocal()) : UnDefType.UNDEF;
+    }
+
+    protected State getLivePictureURLState() {
+        return getLivePictureURL() == null ? UnDefType.UNDEF : toStringType(getLivePictureURL());
+    }
+
+    protected State getLivePictureState() {
+        return getLivePictureURL() == null ? UnDefType.UNDEF : HttpUtil.downloadImage(getLivePictureURL());
+    }
+
+    protected State getLiveStreamState() {
+        return getLiveStreamURL() == null ? UnDefType.UNDEF : new StringType(getLiveStreamURL());

```

Comment:
Please cache `getLiveStreamURL()` to a local variable to it doesn't get recomputed. Also do the same for  the other methods above.

Reply:
done